crates/sui-core/src/consensus_handler.rs

```diff
@@ -63,85 +64,94 @@ impl ConsensusHandler {
 impl ExecutionState for ConsensusHandler {
     /// This function will be called by Narwhal, after Narwhal sequenced this certificate.
     #[instrument(level = "trace", skip_all)]
-    async fn handle_consensus_transaction(
+    async fn handle_consensus_output(
         &self,
         // TODO [2533]: use this once integrating Narwhal reconfiguration
-        consensus_output: &Arc<ConsensusOutput>,
-        consensus_index: ExecutionIndices,
-        serialized_transaction: Vec<u8>,
+        consensus_output: ConsensusOutput,
     ) {
-        let index = Self::update_hash(
-            &self.last_seen,
-            consensus_index.clone(),
-            &serialized_transaction,
-        );
-        let index = if let Some(index) = index {
-            index
-        } else {
-            debug!(
+        let _scope = monitored_scope("HandleConsensusOutputFull");
+        let mut sequenced_transactions = Vec::new();
+        let mut seq = 0;
+
+        for (cert, batches) in consensus_output.batches {
+            let round = cert.header.round;
```

Comment:
This actually needs to be a sub_dag round, not certificate round. `CommittedSubDag::round` can be used for that, and you can evaluate it outside of this loop

Reply:
Oh yeah - cert.header.round is the round of given certificate, while CommittedSubDag::round is the round of leader certificate of the sub dag, subdag can have certificates from different rounds